app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,125 @@
+import { ObservableStore } from '@metamask/obs-store';
+import MetaMetricsController from '../metametrics';
+import {
+  createDataDeletionRegulationTask,
+  fetchDeletionRegulationStatus,
+} from './services/services';
+
+export type DataDeleteDate = string;
+export type DataDeleteRegulationId = string;
+
+export type RegulationId = Record<string, string>;
+export type CurrentRegulationStatus = Record<string, Record<string, string>>;
+export type DeleteRegulationAPIResponse = {
+  data: RegulationId | CurrentRegulationStatus;
+};
+
+export enum DeleteRegulationStatus {
+  FAILED = 'FAILED',
+  FINISHED = 'FINISHED',
+  INITIALIZED = 'INITIALIZED',
+  INVALID = 'INVALID',
+  NOT_SUPPORTED = 'NOT_SUPPORTED',
+  PARTIAL_SUCCESS = 'PARTIAL_SUCCESS',
+  RUNNING = 'RUNNING',
+  UNKNOWN = 'UNKNOWN',
+}
+
+export type MetaMetricsDataDeletionState = {
+  metaMetricsDataDeletionId: DataDeleteRegulationId;
+  metaMetricsDataDeletionDate: DataDeleteDate;
+  metaMetricsDataDeletionStatus?: DeleteRegulationStatus;
+};
+
+const defaultState: MetaMetricsDataDeletionState = {
+  metaMetricsDataDeletionId: '',
+  metaMetricsDataDeletionDate: '',
+  metaMetricsDataDeletionStatus: undefined,
+};
+
+/**
+ * Controller responsible for maintaining
+ * state related to Metametrics data deletion
+ */
+export default class MetaMetricsDataDeletionController {
+  /**
+   * Observable store containing controller data.
+   */
+  store: ObservableStore<MetaMetricsDataDeletionState>;
+
+  private metaMetricsController: MetaMetricsController;
+
+  /**
+   * Constructs a DeleteMetaMetricsData  controller.
+   *
+   * @param options - the controller options
+   * @param options.initState - Initial controller state.
+   * @param options.metaMetricsController
+   */
+  constructor({
+    initState,
+    metaMetricsController,
+  }: {
+    initState: MetaMetricsDataDeletionState;
+    metaMetricsController: MetaMetricsController;
+  }) {
+    this.store = new ObservableStore({
```

Comment:
It looks like you're building an older extension style controller. These are deprecated, we should be using BaseControllerV2 instead to build controllers.